kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -97,94 +97,32 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   def createExecutorPod(executorNum: Int): String = {
     // create a single k8s executor pod.
-    var annotationMap = Map(
-      "pod.beta.kubernetes.io/init-containers" -> raw"""[
-                {
-                    "name": "client-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/client.jar", "$clientJarUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "distro-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/spark.tgz", "$sparkDistUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "setup",
-                    "image": "$sparkDriverImage",
-                    "command": ["./install.sh"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        },
-                        {
-                            "name": "opt",
-                            "mountPath": "/opt"
-                        }
-                    ]
-                }
-            ]""")
-
-
     val labelMap = Map("type" -> "spark-executor")
     val podName = s"$sparkExecutorName-$executorNum"
     var pod = new PodBuilder()
       .withNewMetadata()
       .withLabels(labelMap.asJava)
       .withName(podName)
-      .withAnnotations(annotationMap.asJava)
       .endMetadata()
       .withNewSpec()
       .withRestartPolicy("OnFailure")
+
       .addNewContainer().withName("spark-executor").withImage(sparkDriverImage)
-      .withImagePullPolicy("Always")
-      .withCommand("/opt/spark/bin/spark-class")
+      .withImagePullPolicy("IfNotPresent")
+      .withCommand("/opt/executor.sh")
```

Comment:
My theory on this is it would be preferable to adopt a "convention over configuration" policy on this (that is, document that if you want to spin your own container, you need `driver.sh` and `executor.sh`)
Not that we couldn't use better names.

However, it would not be onerous to make `path/to/driver-script` and `path/to/executor-script` into configuration parameters.